### PR TITLE
Gig UI

### PR DESCRIPTION
**Pull request by bcongdon**
* Can view and create gigs
* Users can claim gigs; gig creators can can accept/reject claims
    * Tied into groot-credits-service to automatically transfer funds
* Admins can delete gigs
* Admins can create admin-tasks that create credits upon accepted claims, rather than debit the creator's account

Todo: 
- [x] Authenticate claim accepting
- [x] Authenticate claim deleting
- [x] Authenticate gig closing

screen shot 2017-03-21 at 4 22 27 pm
screen shot 2017-03-21 at 4 21 29 pm

### CODE REVIEW

File: app/controllers/gigs.js [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+**/
+
+const SERVICES_URL = process.env.SERVICES_URL || 'http://localhost:8000';
+const GROOT_ACCESS_TOKEN = process.env.GROOT_ACCESS_TOKEN || "TEMP_STRING";
+const request = require('request');
+const moment = require('moment');
+const utils = require('../../etc/utils.js');
+
+module.exports = function(app) {
+  app.get('/gigs', function(req, res) {
```
Comment (sameetandpotatoes) [inline app/controllers/gigs.js]: Did we want these under the `intranet` sub-route? Like `/intranet/gigs`?

File: app/controllers/gigs.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        issuer: req.session.student.netid,
+        title: req.body.title,
+        details: req.body.details,
+        credits: req.body.credits,
+        admin_task: req.body.admin_gig
+      },
+      json: true
+    }, function(err, response, body) {
+      if(err || !body || body.error) {
+        req.flash('error', 'Unable to create gig');
```
Comment (sameetandpotatoes) [inline app/controllers/gigs.js]: Can we put `body.error` in the error messages so we can potentially get something more useful out of the error message?

File: app/controllers/gigs.js (hunk 1 above)
Comment (narendasan) [inline app/controllers/gigs.js]: thats probably a better place to put it

Comment (bcongdon) [inline app/controllers/gigs.js]: Sounds good.

File: app/controllers/gigs.js (hunk 2 above)
Comment (bcongdon) [inline app/controllers/gigs.js]: Sure

File: app/controllers/gigs.js [hunk 3]
```diff
@@ -148,28 +148,44 @@ module.exports = function(app) {
...
+      return res.redirect('/intranet/gigs');
     });
   });
-  app.put('/gigs/:gig_id', function(req, res) {
+  app.put('/intranet/gigs/:gig_id', function(req, res) {
     request({
-      url: `${SERVICES_URL}/gigs/${req.params.gig_id}`,
-      method: "PUT",
+      url: `${SERVICES_URL}/gigs/claims/${req.params.claim_id}`,
+      method: "GET",
```
Comment (bcongdon) [inline app/controllers/gigs.js]: This is done to validate that the current user is actually the "owner" of the claim. I didn't want to do the token authentication thing again, and we're implying trust between the services, so the wrapping request is used to get the claim requester before issuing the DELETE/PUT action

Comment (sameetandpotatoes) [inline app/controllers/gigs.js]: Alright, yeah this seems best then.

**Pull request merged**